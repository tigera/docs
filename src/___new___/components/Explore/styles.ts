export default {};

export const howItWorksStyle = {
  width: '100%',
  h: 'auto',
  bg: 'linear-gradient(105.96deg, #0964AD 56.79%, #06547A 100%)',
  flexDirection: 'column',
  pb: '60px',
};

export const stackStyle = {
  justifyContent: 'center',
  h: 'auto',
  alignItems: 'center',
  mb: -2,
  px: [4, 0],
};

export const headerTextStyle = {
  textAlign: 'center',
  pt: 14,
  mb: 4,
};

export const subHeaderTextStyle = (isDarkMode: boolean) => ({
  fontWeight: '600',
  textAlign: 'center',
  color: isDarkMode ? 'tigeraWhite' : 'tigeraBlack',

  mt: 7,
  fontSize: '2xl',
  lineHeight: 9,
});

export const innerTextStyle = (isDarkMode: boolean) => ({
  textAlign: 'center',
  fontSize: 'xl',
  fontWeight: 'normal',
  color: isDarkMode ? 'tigeraWhite' : 'tigeraGrey.800',
  mx: '44px',
  mb: 10,
  mt: 2,
  lineHeight: 8,
});

export const rectangleStyle = (isDarkMode: boolean) => ({
  width: ['100%', '371px'],
  border: '1px',
  borderColor: 'tigeraCloudGrey',
  borderRadius: 'lg',
  bg: isDarkMode ? 'tigeraBlack' : 'tigeraWhite',
  ml: [0, 4],
  mr: [0, 4],
  mt: [4, 4, 4, 4, ''],
  mb: [4, 4, 4, 4, ''],
  pb: '40px',
});

export const actionBoxStyles = {
  mb: 14,
  justifyContent: 'center',
};
