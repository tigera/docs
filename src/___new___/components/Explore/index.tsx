//Possibly to be removed

import * as React from 'react';
import { Box, Flex, Text, SystemStyleObject, Stack, Heading } from '@chakra-ui/react';
import { heading2Styles } from '../styles';
import howItWorksInfo from '../../data/exploreInfo';

import {
  howItWorksStyle,
  innerTextStyle,
  rectangleStyle,
  stackStyle,
  subHeaderTextStyle,
  headerTextStyle,
} from './styles';

interface HowItWorksProps {
  sx?: SystemStyleObject;
  isDarkMode: boolean;

}

const HowItWorks: React.FC<HowItWorksProps> = ({ isDarkMode,...rest }) => (
  <Flex
    sx={howItWorksStyle}
    {...rest}
  >
    <Heading
      as='h2'
      size='md'
      sx={{ ...heading2Styles(true), ...headerTextStyle }}
    >
      Explore the basics...
    </Heading>
    <Stack
      sx={stackStyle}
      direction={['column', 'column', 'column', 'column', 'row', 'row', 'row']}
    >
      {howItWorksInfo.howItWorksInfo.map((info, index) => (
        <Box key={index}>
          <Box
            sx={rectangleStyle(isDarkMode)}
            key={index}
          >
            <Text sx={subHeaderTextStyle(isDarkMode)}>{info.title}</Text>
            {info.description.map((desc, index) => (
              <Text
                key={index}
                sx={innerTextStyle(isDarkMode)}
              >
                {desc}
              </Text>
            ))}
          </Box>
        </Box>
      ))}
    </Stack>
  </Flex>
);

export default HowItWorks;
