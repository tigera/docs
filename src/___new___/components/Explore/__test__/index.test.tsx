import { render } from '@testing-library/react';
import Explore from '../index';

describe('<Explore/>', () => {
  it('should render the Explore component', () => {
    const { asFragment } = render(<Explore isDarkMode={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
