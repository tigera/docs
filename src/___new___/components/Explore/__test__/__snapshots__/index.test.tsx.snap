// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Explore/> should render the Explore component 1`] = `
<DocumentFragment>
  <div
    class="css-1xuj0g6"
  >
    <h2
      class="chakra-heading css-17pfrz0"
    >
      Explore the basics...
    </h2>
    <div
      class="chakra-stack css-1mtxuht"
    >
      <div
        class="css-0"
      >
        <div
          class="css-16184rp"
        >
          <p
            class="chakra-text css-pb7mbv"
          >
            What is Calico Open Source?
          </p>
          <p
            class="chakra-text css-3u4gm1"
          >
            Calico Open Source is a free, self-managed networking and security solution for containers, virtual machines, and native host-based workloads. Calico provides networking, network policy, and IP address management capabilities for your cloud-native applications.
          </p>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-16184rp"
        >
          <p
            class="chakra-text css-pb7mbv"
          >
            What is Calico Enterprise?
          </p>
          <p
            class="chakra-text css-3u4gm1"
          >
            Calico Enterprise is a paid, self-managed security and observability solution for containers and Kubernetes. Calico Enterprise extends Calico Open Sourceâ€™s networking and network security capabilities to offer more advanced security and observability capabilities for organizations running Kubernetes at scale.
          </p>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-16184rp"
        >
          <p
            class="chakra-text css-pb7mbv"
          >
            What is Calico Cloud?
          </p>
          <p
            class="chakra-text css-3u4gm1"
          >
            Calico Cloud is a pay-as-you-go, SaaS application that provides comprehensive container security across the entire container lifecycle (build, deploy, runtime). In addition to being a fully-managed version of Calico Enterprise, Calico Cloud provides container image scanning and more advanced threat detection capabilities.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
